game/state/city/scenery.cpp

```diff
@@ -1486,6 +1488,7 @@ void Scenery::updateFalling(GameState &state, unsigned int ticks)
 {
 	auto fallTicksRemaining = ticks;
 	auto newPosition = currentPosition;
+	state.current_city->map->setViewSurfaceDirty(currentPosition.z);
```

Comment:
Perhaps the setViewSurfaceDirty() should be moved to the TileObject parent class? So that at TileObject add/remove/move it can mark the correct dirty bit? Perhaps have a dirty bit per "layer" (TileMap::layerMap) to 

This will still need special handling for sprites that can change (if you want to handle more than the scenery layer) -
 e.g. doodads/animations changing frame, vehicles/agents/projectiles when they rotate - or just not enable caches for those layers?

Reply:
Yes, its a good architectural approach.
I think only Scenery and BattleMapParts should be drawn to the cache. Thus only these classes can set the cache dirty.

Vehicles, units, projectiles etc should be handled separately because they movable.